## Objectives 
What do we want to know from the NASATLX tests from the participants every after drive. 

1. we want to know how heavy the work load is for every respondent 
- look into comparisons (effort and frustration = )

In [13]:
import pandas as pd
from statsmodels.stats.anova import AnovaRM
from scipy.stats import ttest_rel
import numpy as np 
import seaborn as sns
import matplotlib.pyplot as plt

In [14]:
df = pd.read_csv("cleaned_NASATLX.csv")

display(df)

,Participant,Task_Name,Mental_Demand,Physical_Demand,Temporal_Demand,Performance,Effort,Frustration,Raw_Score
0,8,AH,70,60,40,70,60,80,63.33
1,8,NC,80,70,70,70,70,70,71.67
2,8,AO,80,70,70,70,70,70,71.67
3,8,HO,80,70,70,70,70,70,71.67
4,9,NC,20,40,40,60,60,40,43.33
...,...,...,...,...,...,...,...,...,...
67,11,NC,5,15,5,65,50,0,23.33
68,5,AH,80,80,60,20,20,20,46.67
69,5,NC,80,80,60,10,10,10,41.67
70,5,HO,90,90,70,20,20,20,51.67


In [15]:
print(df.columns)

Index(['Participant', 'Task_Name', 'Mental_Demand', 'Physical_Demand',
       'Temporal_Demand', 'Performance', 'Effort', 'Frustration', 'Raw_Score'],
      dtype='object')


**Running Repeated Measures ANOVA**

Pr > F ; if P <0.05 -> there is a significant effect of cue condition 
Answers the question: Does cue condition affect each workload component? 

- P has no significant effect on cue condition

In [16]:
anova = AnovaRM(
    data=df,
    depvar="Mental_Demand",
    subject="Participant",
    within=["Task_Name"]
).fit()

print(anova)

                 Anova
          F Value Num DF  Den DF Pr > F
---------------------------------------
Task_Name  1.1104 3.0000 51.0000 0.3535



In [17]:
anova = AnovaRM(
    data=df,
    depvar="Physical_Demand",
    subject="Participant",
    within=["Task_Name"]
).fit()

print(anova)

                 Anova
          F Value Num DF  Den DF Pr > F
---------------------------------------
Task_Name  1.4609 3.0000 51.0000 0.2362



In [18]:
anova = AnovaRM(
    data=df,
    depvar="Temporal_Demand",
    subject="Participant",
    within=["Task_Name"]
).fit()

print(anova)

                 Anova
          F Value Num DF  Den DF Pr > F
---------------------------------------
Task_Name  1.4941 3.0000 51.0000 0.2272



In [19]:
anova = AnovaRM(
    data=df,
    depvar="Performance",
    subject="Participant",
    within=["Task_Name"]
).fit()

print(anova)

                 Anova
          F Value Num DF  Den DF Pr > F
---------------------------------------
Task_Name  0.4140 3.0000 51.0000 0.7436



In [20]:
anova = AnovaRM(
    data=df,
    depvar="Effort",
    subject="Participant",
    within=["Task_Name"]
).fit()

print(anova)

                 Anova
          F Value Num DF  Den DF Pr > F
---------------------------------------
Task_Name  0.1845 3.0000 51.0000 0.9065



In [21]:
anova = AnovaRM(
    data=df,
    depvar="Frustration",
    subject="Participant",
    within=["Task_Name"]
).fit()

print(anova)

                 Anova
          F Value Num DF  Den DF Pr > F
---------------------------------------
Task_Name  1.8127 3.0000 51.0000 0.1566



**Comparing Conditions** depending on the cue

In [22]:
anova_raw = AnovaRM(
    data=df,
    depvar="Raw_Score",
    subject="Participant",
    within=["Task_Name"]
).fit()

print(anova_raw)

                 Anova
          F Value Num DF  Den DF Pr > F
---------------------------------------
Task_Name  0.9729 3.0000 51.0000 0.4128



Num DF and Den DF will not change across all comparisons.
- Num DF is degrees of freedom of factor = 3 (k-1)
- Den DF is degrees of freedom of error = 51 (n-1)(k-1)

n = number of participants = 18, supposedly 20 
k = number of conditions (alerts) = 4 alerts 

F-Value measures: 
Numerator = variability between tasks
Denominator = variability within subjects (error) 
- Large F = condition means are very different (significant effect) 
    - critical F ~ 2.79, so if F>= 2.8 (significant)
- Small F = differene between means are small relative to noise 

Pr > F (p-value)
- null hypothesis true or not
- p > 0.05

**There is no significant difference in NASA-TLX raw score workload across all 4 alert types.**

The results indicate no statistically significant difference in raw score F = 0.97, p = 0.41. 

The smallest F Value is effort with a 0.18, while the highest is Frustration with 1.81. This also translates to the p-value with effort having 0.91 and frustration having 0.16. 
 Even if frustration is larger than the rest, 1.81 is still smaller than the critical value being 2.79. 

 All of the dimensions in NASA-TLX are below the critical value which tells us to **Fail to reject the Null Hypothesis**. All p-values were greater than the significance level of 0.05, that tells us that there were no statistically significant differences in the workload across the four alert types for any NASA-TLX dimension. Additionally, all the F-values were bellow the critical value (2.79). These results suggest that despite the alert type, it did not affect the participant's percieve cognitive workload. 

### Baseline Comparisons
Paired t-test on baseline vs. alerts

In [23]:
NC = df[df["Task_Name"] == "NC"]
AO = df[df["Task_Name"] == "AO"]
HO = df[df["Task_Name"] == "HO"]
AH = df[df["Task_Name"] == "AH"]

In [25]:
print("NC vs AO:", ttest_rel(NC["Raw_Score"], AO["Raw_Score"]))
print("NC vs HO:", ttest_rel(NC["Raw_Score"], HO["Raw_Score"]))
print("NC vs AH:", ttest_rel(NC["Raw_Score"], AH["Raw_Score"]))

NC vs AO: TtestResult(statistic=-0.7877286199864711, pvalue=0.4417011121253792, df=17)
NC vs HO: TtestResult(statistic=-1.4459321981282358, pvalue=0.16638070651284692, df=17)
NC vs AH: TtestResult(statistic=-1.337370677023179, pvalue=0.19872127009632867, df=17)


NC vs. AO 
- P-value is 0.44, this is not statistically significant (meaning audio only does not affect cognative load) 
- statistic = -0.78 difference (subtraction) between the two conditions (meaning there are very small difference (pagkakaiba) between the two alert types)
- this has the highest among the three alerts, which signifies that this has the least added cognative load. 

NC vs. HO
- p-value is 0.17, this is not significant, which tells us that the haptic vibrations does affect cognative load. 
- statistic = -1.45, this is a small difference between the baseline and haptics. 
- This has the closest to 0.05 among the 3 alerts as well as the highest difference, which can tell us that it has the highest probability to add to cognative load. Restassured, haptics do not add to the cognitive load, this just means it is the most probable one among the three.

NC vs. AH 
- p-value is 0.20, this is not significant, which tells us
- 
- The combined audio and haptic, is in between the audio only and haptic only in terms of statistics that affect the cognitive load. 

All in all, NASA-TLX workload in baseline is not significantly different from any alert condition. This is good, because this proves the baseline's scores were not very different from any of the alerts. This further proves that the alerts did not add cogintive load when triggered. 
- p-values across the baseline comparisons are > 0.05. 
- statistic difference is considerable small as values are near 0. 
- The Audio only has the least added cognitive load, followed by combined audio and haptic, while haptic only is the last causing most cognitive load. 

### Descriptive Statistics: Means and Standard Deviation

In [26]:
df.groupby("Task_Name")["Raw_Score"].mean()

Task_Name
AH    35.786111
AO    34.630556
HO    36.944444
NC    32.407222
Name: Raw_Score, dtype: float64

In [27]:
MD_sum = df.groupby("Task_Name")["Mental_Demand"].mean().round(2)
PD_sum = df.groupby("Task_Name")["Physical_Demand"].mean().round(2)
TD_sum = df.groupby("Task_Name")["Temporal_Demand"].mean().round(2)
P_sum = df.groupby("Task_Name")["Performance"].mean().round(2)
E_sum = df.groupby("Task_Name")["Effort"].mean().round(2)
F_sum = df.groupby("Task_Name")["Frustration"].mean().round(2)

In [29]:
conditions = ["NC", "AO", "HO", "AH"]

for condition in conditions:
    subset = df[df["Task_Name"] == condition]["Raw_Score"]
    mean = subset.mean()
    sd = subset.std()
    print(f"{condition}: M = {mean:.2f}, SD = {sd:.2f}")

NC: M = 32.41, SD = 17.47
AO: M = 34.63, SD = 17.22
HO: M = 36.94, SD = 19.37
AH: M = 35.79, SD = 19.58


Worload scores are similar across the 4 alert types. Although AO, HO, and AH have a slighty higher mean scores than NC, the difference is too small to be statistically significant. This further supports that the alert modalities did not increase or decrease the perceived workload compared to baseline. 

## Summary

Overall, the alert type did not significantly affect the perceived cognitive load. However, the study only included a sample size of around 20 participants. This relatively small sample size may have limited the statistical power of the analysis and reduced the ability to detect potentially significant differences.